packages/components/icon/src/Icon.tsx

```diff
@@ -0,0 +1,142 @@
+import { css, cx } from 'emotion';
+import React, { forwardRef } from 'react';
+import tokens from '@contentful/f36-tokens';
+import { Primitive } from '@contentful/f36-core';
+import type {
+  CommonProps,
+  ComponentVariant,
+  Simplify,
+} from '@contentful/f36-core';
+import type {
+  ComponentType,
+  ExoticComponent,
+  ReactElement,
+  SVGAttributes,
+} from 'react';
+
+export type IconComponent = ExoticComponent<any> | ComponentType<any>;
+
+type IconVariant = Simplify<ComponentVariant | 'muted' | 'white'>;
+
+type AsOrChildren =
+  | {
+      as: ExoticComponent<any> | ComponentType<any>;
+      children?: never;
+    }
+  | {
+      as?: never;
+      children: ReactElement | ReactElement[];
+    };
+
+type IconInternalProps = {
+  /**
+   * The name prop is used to check if "trimmed" styling should be applied,
+   * which sets the `width` to `auto`.
+   */
+  name?: string;
```

Comment:
Can we just pass `width: auto` to all icons that are have to trimmed?

I think checking if name starts with "trimmed" isn't a good API for this.

Reply:
You mean like a static list of which icons we want to trim? Most icons with a trimmed width also have a non-trimmed version.

Reply:
You can pass `trimmed: true` in icon files (where `generateIcon` is used).